### ISSUE TITLE
Could not find expected rustdoc output for excluded crate

### ISSUE THREAD

**Issue by Darksonn** [obi1kenobi/cargo-semver-checks#380]
We're getting the following error in Tokio's CI:
```
     Parsing benches v0.0.0 (current)
     Parsing benches v0.0.0 (current)
     Parsing examples v0.0.0 (current)
     Parsing examples v0.0.0 (current)
     Parsing stress-test v0.1.0 (current)
     Parsing stress-test v0.1.0 (current)
     Parsing tests-build v0.1.0 (current)
     Parsing tests-build v0.1.0 (current)
     Parsing tests-build v0.1.0 (baseline)
    Checking tests-build v0.1.0 -> v0.1.0 (no change)
   Completed [   0.032s] 41 checks; 41 passed, 0 unnecessary
     Parsing tests-integration v0.1.0 (current)
     Parsing tests-integration v0.1.0 (current)
     Parsing tests-integration v0.1.0 (baseline)
    Checking tests-integration v0.1.0 -> v0.1.0 (no change)
   Completed [   0.032s] 41 checks; 41 passed, 0 unnecessary
     Parsing tokio v1.25.0 (current)
     Parsing tokio v1.25.0 (current)
     Parsing tokio v1.25.0 (baseline)
    Checking tokio v1.25.0 -> v1.25.0 (no change)
   Completed [  14.871s] 41 checks; 41 passed, 0 unnecessary
     Parsing tokio-macros v1.8.2 (current)
     Parsing tokio-macros v1.8.2 (current)
     Parsing tokio-macros v1.8.2 (baseline)
    Checking tokio-macros v1.8.2 -> v1.8.2 (no change)
   Completed [   0.033s] 41 checks; 41 passed, 0 unnecessary
     Parsing tokio-stream v0.1.12 (current)
     Parsing tokio-stream v0.1.12 (current)
     Parsing tokio-stream v0.1.12 (baseline)
    Checking tokio-stream v0.1.12 -> v0.1.12 (no change)
   Completed [   0.183s] 41 checks; 41 passed, 0 unnecessary
     Parsing tokio-test v0.4.2 (current)
     Parsing tokio-test v0.4.2 (current)
     Parsing tokio-test v0.4.2 (baseline)
    Checking tokio-test v0.4.2 -> v0.4.2 (no change)
   Completed [   0.049s] 41 checks; 41 passed, 0 unnecessary
     Parsing tokio-util v0.7.7 (current)
     Parsing tokio-util v0.7.7 (current)
     Parsing tokio-util v0.7.7 (baseline)
    Checking tokio-util v0.7.7 -> v0.7.7 (no change)
   Completed [   0.878s] 41 checks; 41 passed, 0 unnecessary
Error: Could not find expected rustdoc output for `benches`: /home/runner/work/tokio/tokio/target/semver-checks/local-benches-0_0_0/target/semver-checks/target/doc/benches.json
```
Note that benches is the first crated excluded using `--exclude` in our command:

https://github.com/tokio-rs/tokio/blob/d7b7c6131774ab631be6529fef3680abfeeb4781/.github/workflows/ci.yml#L233-L254

I can reproduce the behavior when running the command in the above `ci.yml` script locally. I tried reproducing it in another workspace, but I wasn't able to trigger it there.

**Comment by obi1kenobi** [comment]
Thanks for the bug report, this indeed looks like a bug. I'm looking into it.

Does your configuration maybe have a custom target directory defined, either via a `.cargo/config.toml` file or one of the env vars that set that? It's possible that might be related.

**Comment by Darksonn** [comment]
No, we use the default of having the target directory in the workspace directory.

**Comment by obi1kenobi** [comment]
Assuming I got the fix in #381 right, I think you won't need all the `--exclude` options. I think they were needed before this because the CLI wasn't behaving correctly in workspaces.

**Comment by obi1kenobi** [comment]
I'm releasing v0.18.3 and yanking the previous v0.18.x versions because I found an unfortunate bug where CLI argument misparsing could cause false-negatives.

v0.18.3 is wrapping up the publish process right now and should be available in just a few minutes.
